### ISSUE TITLE
linkState eventPath could be documented better

### ISSUE THREAD

**Issue by jakearchibald** [developit/preact#376]
https://preactjs.com/guide/linked-state#linked-state-to-the-rescue

I love `linkState`, but the it's unclear from the page above what the second arg does.

It might be useful to explain it in terms of radio buttons:

``` js
class Foo extends Component {
  render({}, {yesChecked, noChecked}) {
    return <div>
      <label>
        <input
          type="radio" name="yes-or-no" checked={yesChecked}
          onChange={this.linkState('yesChecked')}
        />
        Yes
      </label>
      <label>
        <input
          type="radio" name="yes-or-no" checked={noChecked}
          onChange={this.linkState('noChecked')}
        />
        No
      </label>
    </div>;
  }
}
```

The above doesn't work as expected. If the user clicks "no", `noChecked` becomes `true` but `yesChecked` remains `true`, as `onchange` is not fired on the other radio.

`linkState`'s second argument helps here. It lets you provide a path on the event object to use as the linked value:

``` js
class Foo extends Component {
  render({}, {answer}) {
    return <div>
      <label>
        <input
          type="radio" name="yes-or-no" value="yes" checked={answer == 'yes'}
          onChange={this.linkState('answer', 'target.value')}
        />
        Yes
      </label>
      <label>
        <input
          type="radio" name="yes-or-no" value="no" checked={answer == 'no'}
          onChange={this.linkState('answer', 'target.value')}
        />
        No
      </label>
    </div>;
  }
}
```

**Comment by developit** [comment]
I agree, definitely under-explained. I've always found radio buttons a bit awkward to use in react-style code, but your example seems to smooth that over nicely.  I think it's worth adding your second example along with a more detailed explanation.

I can add that, or if you're wanting to submit a PR the content is here:
https://github.com/developit/preact-www/blob/master/content/guide/linked-state.md

**Comment by jakearchibald** [comment]
Happy for you to add it, or I can pick it up in a couple of weeks when deadlines die down a bit.

**Comment by developit** [comment]
heh, I'll put something in there for now.

**Comment by pl12133** [comment]
I took a shot at this in https://github.com/pl12133/preact-www/commit/f5d548288739754c2a549532c1f469119a0ee8a0. Rather than leaving out the optional argument, I think it should be explicitly shown and _then_ tell the reader that it is an optional argument. I also added the example @jakearchibald posted as an "easy controlled input" with linked state.

@developit I was wondering if this statement is still correct when using the second argument:
> Multiple calls to `linkState(name)` with the same `name` are cached, so there is essentially no performance penalty.

Any other ways we can make `linkState` even more clear?
